src/adapters/MeetingsJSONAdapter.js

```diff
@@ -44,5 +69,66 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
 
       observer.complete();
     });
+
+    const muteEvent$ = fromEvent(document, MUTE_AUDIO_CONTROL).pipe(
+      filter((event) => event.detail.ID === ID),
+      map((event) => event.detail)
+    );
+
+    return concat(json$, muteEvent$);
+  }
+
+  /**
+   * Returns an observable that emits the display data of a meeting control.
+   *
+   * @param {string} ID ID of the meeting for which to update display
+   * @returns {Observable<MeetingControlDisplay>}
+   * @memberof MeetingJSONAdapter
+   * @private
+   */
+  muteAudioControl(ID) {
+    return fromEvent(document, MUTE_AUDIO_CONTROL).pipe(
+      filter((event) => event.detail.ID === ID),
+      map((event) => {
+        const isMuted = event.detail.localAudio.muted;
+        let current = null;
+
+        if (isMuted) {
+          current = {
+            ID: MUTE_AUDIO_CONTROL,
+            icon: 'microphone-muted',
+            tooltip: 'Unmute',
+            state: MeetingControlState.ACTIVE,
+          };
+        } else {
+          current = {
+            ID: MUTE_AUDIO_CONTROL,
+            icon: 'microphone',
+            tooltip: 'Mute',
+            state: MeetingControlState.INACTIVE,
+          };
+        }
+
+        return current;
+      })
+    );
+  }
+
+  /**
+   * Toggles muting the local audio media stream track.
+   * Used by "mute-audio" meeting control.
+   *
+   * @param {string} ID  ID of the meeting for which to mute local audio.
+   * @memberof MeetingsJSONAdapter
+   * @private
+   */
+  toggleMuteAudio(ID) {
```

Comment:
As discussed before, we've seen that "toggling" async media operations causes user confusion because multiple clicks causing toggles to fire repeatedly. I know with the json adapter, we don't have async operations, but I'd prefer to not have this as a precedent. 

A more direct approach with the action accepting the desired state would be better:
`meetingsJSONAdapter.muteAudio(meetingID, true)`

Reply:
We discussed this, but putting here for the record.
It is not possible for the action to take a second parameter, as there is no way for the component to know what the "right" state in a generic manner (e.g. any control with different and potentially different number of states).
The async operations are an implementation detail that shouldn't bubble on design, so we came up with solutions for adapters like the SDK:
* Set the control to `DISABLE` while we get acknowledgement from the server and then set it to the appropriate state
* Throttle or ignore subsequent request until we get server acknowledgement